File: App/Resources/AlgorithmDetails/optimizedgnomesort/optimizedgnomesort.java
import java.util.Arrays;

public class optimizedgnomesort {
  public static void sort(int arr[]) {
    for (int i = 1; i < arr.length; i++) {
      int pos = i;
      while (pos > 0 && arr[pos - 1] > arr[pos]) {
        int temp = arr[pos - 1];
        arr[pos - 1] = arr[pos];
        arr[pos] = temp;
        pos--;
      }
    }
  }

  public static void main(String[] args) {
    int[] array = new int[]{0, 39, 21, 62, 91, 77, 14, 23, 90, 69, 51, 81, 68, 83, 32, 56};
    sort(array);
    System.out.println(Arrays.toString(array));
  }
}

File: App/Resources/AlgorithmDetails/optimizedgnomesort/optimizedgnomesort.go
package main

import (
	"fmt"
)

func sort(arr []int) []int {
	for i := 1; i < len(arr); i++ {
		pos := i
		for pos > 0 && arr[pos-1] > arr[pos] {
			arr[pos-1], arr[pos] = arr[pos], arr[pos-1]
			pos--
		}
	}
	return arr
}

func main() {
	array := []int{0, 39, 21, 62, 91, 77, 14, 23,
		90, 69, 51, 81, 68, 83, 32, 56}
	fmt.Println(sort(array))
}
